frontend/src/components/pages/your_clubs/CreateNewClub.tsx

```diff
@@ -0,0 +1,242 @@
+import { makeStyles } from "@material-ui/core/styles";
+import { ServiceContext } from "../../contexts/contexts";
+import TextField from '@material-ui/core/TextField';
+import { Theme } from "@material-ui/core/styles";
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    bookField: {
+      justifyContent: 'center',
+      margin: theme.spacing(2),
+    },
+    successButton: {
+      marginLeft: theme.spacing(2),
+      marginBottom: theme.spacing(2),
+    },
+  }),
+);
+
+interface CreateNewClubWindowProps {
+  isCreatingNewClub: boolean,
+  closeWindow(successfulCreation: boolean): void,
+}
+
+export function CreateNewClubWindow(props: CreateNewClubWindowProps) {
+  const classes = useStyles();
+  const [club, setClub] = React.useState<ClubProps|undefined>(undefined);
+  const [book, setBook] = React.useState<BookProps|undefined>(undefined);
+  const [missingField, setMissingField] = React.useState<boolean>(false);
+  const [submitSuccess, setSubmitSuccess] = React.useState<boolean>(false);
+  const [contentWarningsDisplay, setContentWarningsDisplay] =
+    React.useState<string|undefined>(undefined);
+
+  const contextServices = React.useContext(ServiceContext);
+  const yourClubsHandlerService = contextServices.yourClubsHandlerService;
+
+  const handleClubNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setClub({...club, name: event.target.value});
+  }
+
+  const handleClubDescriptionChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setClub({...club, description: event.target.value});
+  }
+
+  const handleClubContentWarningsChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    /*
+     * User should see newline characters in their repsonse,
+     * but backend must be stored as a list of strings.
+     */
+    setContentWarningsDisplay(event.target.value);
+    const contentWarningsArray = (event.target.value).split('\n');
+    setClub({...club, contentWarnings: contentWarningsArray});
+  }
+
+  const handleBookTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setBook({...book, title: event.target.value});
+    setClub({...club, currentBook: book});
+  }
+
+  const handleBookAuthorChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setBook({...book, author: event.target.value});
+    setClub({...club, currentBook: book});
+  }
+
+  const handleBookIsbnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setBook({...book, isbn: event.target.value});
+    setClub({...club, currentBook: book});
+  }
+
+  const handleCloseCreate = (successfulCreation: boolean) => {
+    setClub(undefined);
+    setBook(undefined);
+    setContentWarningsDisplay(undefined);
+    setMissingField(false);
+    setSubmitSuccess(false);
+    props.closeWindow(successfulCreation);
+  }
+
+  const handleClubSubmission = async () => {
+    if (!(club?.name && club?.description && book?.title && book?.author)) {
```

Comment:
nit: This would be cleaner as a disjunction (!X || !Y) rather than a conjunction (!(X && Y)). Since you're a math guy, see De Morgan's laws.